libs/@blockprotocol/graph/src/types/subgraph/vertices.ts

```diff
@@ -114,21 +115,23 @@ export const isEntityVertexId = (
     /** @todo - is it fine to just check that versionId is string, maybe timestamp if we want to lock it into being a
      *    timestamp?
      */
-    !isOntologyTypeRecordId(vertexId)
+    !isOntologyTypeVertexId(vertexId)
   );
 };
 
-export type OntologyVertices = {
-  [typeBaseUri: BaseUri]: {
-    [typeVersion: number]: OntologyVertex;
-  };
-};
-
-export type KnowledgeGraphVertices<Temporal extends boolean> = {
-  [entityId: EntityId]: {
-    [entityVersion: EntityRevisionId]: KnowledgeGraphVertex<Temporal>;
-  };
-};
+export type OntologyVertices = Record<
+  BaseUri,
+  Record<OntologyTypeRevisionId, OntologyVertex>
```

Comment:
It will probably be useful to have separate types for each level in these maps at some point. It would make it a bit easier to write typeguards like this. Not suggesting any action now, we can add them as and when we need them.
```ts
const isEntityVersionObject = <Temporal extends boolean>(
  versionObject:
    | Record<EntityRevisionId, KnowledgeGraphVertex<Temporal>>
    | Record<OntologyTypeRevisionId, OntologyVertex>,
): versionObject is Record<
  EntityRevisionId,
  KnowledgeGraphVertex<Temporal>
> => {
  const sampleVersion = Object.values(versionObject)[0];
  if (!sampleVersion) {
    throw new Error("version object is empty");
  }
  return isEntityVertex(sampleVersion);
};
```

Reply:
We could definitely consider this at some point :)